x-pack/filebeat/input/netflow/input.go

```diff
@@ -83,171 +227,72 @@ func (w *logDebugWrapper) Write(p []byte) (n int, err error) {
 	return n, nil
 }
 
-// NewInput creates a new Netflow input
-func NewInput(
-	cfg *conf.C,
-	connector channel.Connector,
-	context input.Context,
-) (input.Input, error) {
-	initLogger.Do(func() {
-		logger = logp.NewLogger(inputName)
-	})
-	out, err := connector.Connect(cfg)
-	if err != nil {
-		return nil, err
-	}
+// stop stops the netflow input
+func (n *netflowInput) stop() {
+	n.mtx.Lock()
+	defer n.mtx.Unlock()
 
-	config := defaultConfig
-	if err = cfg.Unpack(&config); err != nil {
-		out.Close()
-		return nil, err
+	if !n.started {
+		return
 	}
 
-	var customFields []fields.FieldDict
-	for _, yamlPath := range config.CustomDefinitions {
-		f, err := LoadFieldDefinitionsFromFile(yamlPath)
-		if err != nil {
-			return nil, fmt.Errorf("failed parsing custom field definitions from file '%s': %w", yamlPath, err)
-		}
-		customFields = append(customFields, f)
+	if n.udp != nil {
+		n.udp.Stop()
 	}
-	decoder, err := decoder.NewDecoder(decoder.NewConfig().
-		WithProtocols(config.Protocols...).
-		WithExpiration(config.ExpirationTimeout).
-		WithLogOutput(&logDebugWrapper{Logger: logger}).
-		WithCustomFields(customFields...).
-		WithSequenceResetEnabled(config.DetectSequenceReset).
-		WithSharedTemplates(config.ShareTemplates))
-	if err != nil {
-		return nil, fmt.Errorf("error initializing netflow decoder: %w", err)
-	}
-
-	input := &netflowInput{
-		outlet:           out,
-		internalNetworks: config.InternalNetworks,
-		forwarder:        harvester.NewForwarder(out),
-		decoder:          decoder,
-		logger:           logger,
-		queueSize:        config.PacketQueueSize,
-	}
-
-	input.udp = udp.New(&config.Config, input.packetDispatch)
-	return input, nil
-}
-
-func (p *netflowInput) Publish(events []beat.Event) error {
-	for _, evt := range events {
-		p.forwarder.Send(evt)
-	}
-	return nil
-}
 
-// Run starts listening for NetFlow events over the network.
-func (p *netflowInput) Run() {
-	p.mutex.Lock()
-	defer p.mutex.Unlock()
-
-	if !p.started {
-		logger.Info("Starting UDP input")
-
-		if err := p.decoder.Start(); err != nil {
-			logger.Errorw("Failed to start netflow decoder", "error", err)
-			p.outlet.Close()
-			return
-		}
-
-		p.queueC = make(chan packet, p.queueSize)
-		err := p.udp.Start()
-		if err != nil {
-			logger.Errorf("Error running harvester: %v", err)
-			p.outlet.Close()
-			p.decoder.Stop()
-			close(p.queueC)
-			return
+	if n.decoder != nil {
+		if err := n.decoder.Stop(); err != nil {
+			n.logger.Info("Error stopping decoder", "error", err)
 		}
+	}
 
-		go p.recvRoutine()
-		// Only the first active input launches the stats thread
-		if aliveInputs.Inc() == 1 && logger.IsDebug() {
-			go p.statsLoop()
+	if n.client != nil {
+		if err := n.client.Close(); err != nil {
+			n.logger.Info("Error closing beat client", "error", err)
```

Comment:
[Question]
Same thing here: Why not level error? What are the consequences of this decoder failing to stop?